subsys/zbus/Kconfig

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2022 Rodrigo Peixoto <rodrigopex@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig ZBUS
+	bool "Zbus support"
+	help
+	  Enables support for Zephyr message bus.
+
+if ZBUS
+
+config ZBUS_STRUCTS_ITERABLE_ACCESS
+	bool "Zbus iterable sections support."
+	depends on !XTENSA && !ARCH_POSIX
+	default y
+
+config ZBUS_RUNTIME_OBSERVERS_REGISTRATION
+	bool "Zbus runtime observer registration."
+	default n
+
+config ZBUS_RUNTIME_OBSERVERS_POOL_SIZE
+	int "The size of the runtime observers pool."
+	depends on ZBUS_RUNTIME_OBSERVERS_REGISTRATION
+	default 4
+	help
+	  It applies the Object Pool Pattern, where the objects in the pool are pre-allocated and can be used
+	  and recycled after use. The technique avoids dynamic allocation and allows the code to increase the
+	  number of observers by only changing a configuration.
+
+config ZBUS_CHANNEL_USER_DATA
```

Comment:
How about getting rid of this option and using `(CONFIG_ZBUS_CHANNEL_USER_DATA_SIZE > 0)` to check if the `user_data` field is present?